From 43eebcc1ae03260d722ecfc837af05ac1bc47fa0 Mon Sep 17 00:00:00 2001
From: Tejas Upadhyay <tejas.upadhyay@xilinx.com>
Date: Wed, 27 Jun 2018 02:52:06 -0700
Subject: [PATCH 0379/1851] drm: xlnx: zynqmp_disp: handle software vblank

commit ec903a7e428954b1290ed5ef621c391cd8eb7821 from
https://github.com/Xilinx/linux-xlnx.git

During crtc begin/enable/disable, software vblank should
be on/off accordingly.

Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Tejas Upadhyay <tejas.upadhyay@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index 68d34dfc4d4e..47a40bf83c19 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -2864,6 +2864,7 @@ zynqmp_disp_crtc_atomic_disable(struct drm_crtc *crtc,
 	zynqmp_disp_clk_disable(disp->pclk, &disp->pclk_en);
 	zynqmp_disp_plane_disable(crtc->primary);
 	zynqmp_disp_disable(disp, true);
+	drm_crtc_vblank_off(crtc);
 	pm_runtime_put_sync(disp->dev);
 }
 
@@ -2877,6 +2878,7 @@ static void
 zynqmp_disp_crtc_atomic_begin(struct drm_crtc *crtc,
 			      struct drm_crtc_state *old_crtc_state)
 {
+	drm_crtc_vblank_on(crtc);
 	/* Don't rely on vblank when disabling crtc */
 	spin_lock_irq(&crtc->dev->event_lock);
 	if (crtc->primary->state->fb && crtc->state->event) {
-- 
2.31.1

